core/tps.service.js

```diff
@@ -246,11 +285,38 @@ async function remove_tp_reservation(reservation_id) {
             }
         });
         await dbsrv.mongo_events().insertOne({ 'owner': 'auto', 'date': new Date().getTime(), 'action': 'close reservation for ' + reservation.owner , 'logs': [] });
+    } catch (error) {
+        logger.error(error);
+    }
+}
 
+async function extend_tp_reservation(reservation_id, extension) {
+    logger.info('Extend reservation', reservation_id);
+    let reservation = await dbsrv.mongo_reservations().findOne({'_id': reservation_id});
```

Comment:
You already pulled the reservation from DB in the api route, so maybe pass it here and reuse it instead of re-pulling it from the ID

Reply:
In which case shall I also make the same change for '/tp/:id/reserve/now' and '/tp/:id/reserve/stop'?
They both also pull a reservation from the DB in the api route, pass the id to a service function that then re-pulls the reservation from the id.

Reply:
Sure, might as well save a few DB queries